### DISCUSSION TITLE
Any way to fake an action?

### DISCUSSION THREAD

**Discussion by edalzell** [lorisleiva/laravel-actions#229] [Q&A]
I have an event that has 3 listeners, all actions.

I'm trying to test the event/listeners in isolation, so I need to ignore/fake the other two.

I'd love to have something like `ActionA::fake()`, and have that not run the action. Or a `Actions::fake()`, like `Events`.

Is there a way to do this, without having to mock all the calls?

**Comment by lorisleiva** [comment]
Hey Erin, whilst I've not tried that particular scenario before, I believe if you use `ActionA::mock()` in your test, it should always resolve the mock instance from the container instead of the actual action (even when used as a listener). Let me know if that does the trick.

**Reply by edalzell** [reply]
If you mock you have to define expectations, I’d rather not do that.

**Comment by Z3d0X** [comment]
You could do:
```
ActionA::shouldRun()->once()
```
